libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/RoutesObserver.kt

```diff
@@ -30,10 +30,12 @@ fun interface RoutesObserver {
  * @param navigationRoutes list of currently maintained routes
  * @param ignoredRoutes list of alternative routes that were ignored
  *  because they are invalid for navigation. See [IgnoredRoute] for details.
+ * @param initialLegIndex the initial leg index of the primary route.
  * @param reason why the routes have been updated (re-route, refresh route, and etc.)
  */
 class RoutesUpdatedResult internal constructor(
     val navigationRoutes: List<NavigationRoute>,
+    val initialLegIndex: Int,
```

Comment:
how this number is supposed to be used? Should user always pass it to UI? What if route is refreshed on the second leg, will UI understand that it doesn't need to highlight initial leg as it's passed already?

Reply:
@dzinad , as far as I remember from our conversation, you wanted to rename `initialLegIndex` to current leg index, didn't you?

Reply:
But in this case we will need to also update `currentLegIndex` on route leg change

Reply:
Sharing more thoughts on this topic. Consider following case: user leaves current screen, but navigation keeps leaving, for example there is a running foreground service. When user enters the same screen, screens logic should be able to restore state based on navigation state. If user is on the second leg already, UI will highlight a wrong leg.

Reply:
> When user enters the same screen, screens logic should be able to restore state based on navigation state. If user is on the second leg already, UI will highlight a wrong leg.

Why?
Do you mean that the user will register `RoutesObserver` in `onStart` and potentially receive old data?